# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/MahboobAli1/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [4]:
# Lane: Refresh / Content Opportunity Scoring

# I chose this lane because it focuses on helping content teams prioritize which pages should be reviewed first. Instead of simply predicting whether a page is declining, the goal is to rank pages based on evidence so that reviewers spend their limited time on the highest-value opportunities. This matches the internship’s focus on decision support rather than automation.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [5]:
# Research Question: Which content pages should be reviewed first for refresh, expansion, or improvement?

# Decision: The work improves the decision of which pages deserve review first.

# Action: A content or SEO reviewer can refresh, expand, rewrite, monitor, or protect the page based on the recommendation.

# Unit of Analysis: One content page.

# Output: A ranked opportunity score (0–100) or a probability that the page is at risk of decline.

# Cost of a Wrong Recommendation: If the system recommends the wrong page, the team may waste time reviewing a low-value page. If it misses an important page, the business may lose traffic, clicks, or future growth opportunities.

# Why ML can help: Large content inventories contain thousands of pages, making manual prioritization difficult. Machine learning can combine multiple observable signals to help rank pages that deserve attention first.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
!git clone https://github.com/MahboobAli1/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 132, done.
remote: Counting objects: 100% (132/132), done.
remote: Compressing objects: 100% (96/96), done.
remote: Total 132 (delta 45), reused 86 (delta 20), pack-reused 0 (from 0)
Receiving objects: 100% (132/132), 1.84 MiB | 7.64 MiB/s, done.
Resolving deltas: 100% (45/45), done.


In [18]:
%cd /content/flyrank-ml-internship

/content/flyrank-ml-internship


In [23]:
!pwd
!ls data/raw

/content/flyrank-ml-internship
content_refresh_anonymized.csv


In [27]:
!pwd

/content/flyrank-ml-internship


In [28]:
!ls

AGENTS.md  DATA_USE.md	LICENSE    README.md	     SETUP.md	 work
CLAUDE.md  docs		notebooks  requirements.txt  skills
data	   GUIDE.md	outputs    scripts	     submission


In [29]:
!ls data

raw


In [30]:
!ls data/raw

content_refresh_anonymized.csv


In [31]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

In [33]:


import pandas as pd



df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

df.head()

print(df.shape)
df["trend_direction"].value_counts()

df["impressions_90d"].describe()

df["content_type"].value_counts()

(30000, 44)


,count
content_type,
keyword article,27207
feedly article,2096
comparison article,697


## Quick Look at the Data

The starter dataset contains **30,000 content pages** and **44 features**, providing a sufficiently large dataset for exploring content opportunity scoring.

The dataset includes three main content types:
- **Keyword articles:** 27,207 pages (about 90.7%)
- **Feedly articles:** 2,096 pages (about 7.0%)
- **Comparison articles:** 697 pages (about 2.3%)

These numbers show that the dataset is heavily focused on keyword-based content, which makes it suitable for studying how to prioritize pages for review and refresh. The large number of pages also suggests that manual prioritization would be time-consuming, making machine learning a useful approach for decision support.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [34]:
# ## Careful Words: What I Can and Can't Claim

# This project aims to identify content pages that appear to have strong refresh opportunities based on observed patterns in the available data. The results will be **directional** and intended for **decision support**, helping content teams prioritize which pages to review first.

# This project will **not** prove that refreshing a page will increase traffic or improve rankings. The dataset is observational, so it cannot establish cause-and-effect relationships. It also does **not** predict Google's search algorithm or guarantee future search performance. Final decisions should always include human review and domain expertise.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.